File: Arrays101/414. Third Maximum Number.py
class Solution:
    def thirdMax(self, nums: List[int]) -> int:
        if(len(nums)<3): return max(nums)
            
        first,second, third = float('-inf'), float('-inf'), float('-inf')
        for x in nums:
            if(x>first):
                third,second,first = second,first,x
            elif(first>x>second):
                third,second = second,x
            elif(second>x>third):
                third = x
        if (third != float('-inf')):return third
        return first
        
        

File: Arrays101/414. Third Maximum Number.java
class Solution {
    public int thirdMax(int[] nums) {
        //if(nums.length < 2) return nums[0];
        Integer first = null, second = null, third = null;
        for (Integer n : nums) {
            if (n.equals(first) || n.equals(second) || n.equals(third)) continue;
            if (first == null || n > first) {
                third = second;
                second = first;
                first = n;
            } else if (second == null || n > second) {
                third = second;
                second = n;
            } else if (third == null || n > third) {
                third = n;
            }
        }
        return third == null ? first : third;

    }
}